packages/core/src/tools/codegen/utils.ts

```diff
@@ -48,179 +107,234 @@ const netlifyDeploy = () => !!process.env.NETLIFY && !!process.env.BUILD_ID;
-    baseUrl: appPath,
-  };
+  // 1) Load tsconfig (simple + safe). Use appPath as baseUrl fallback.
+  const compilerOptions = loadCompilerOptions(appPath);
+
+  // Read & parse the map file to an AST (SourceFile)
+  const mapText = fs.readFileSync(mapPath, 'utf8');
+  const source = ts.createSourceFile(mapPath, mapText, ts.ScriptTarget.Latest, true);
 
-  const tsHost = ts.createCompilerHost(cliCompilerOptions, true);
-  const componentMapSourceFile = tsHost.getSourceFile(
-    componentMapFullPath,
-    ts.ScriptTarget.Latest,
-    (msg) => {
-      throw new Error(`Failed to parse ${componentMapFullPath}: ${msg}`);
+  // 1) Collect namespace imports: Identifier -> module specifier string
+  const nameSpaceImports = new Map<string, string>();
+  source.forEachChild((node) => {
+    if (
+      ts.isImportDeclaration(node) &&
+      node.importClause &&
+      node.importClause.namedBindings &&
+      ts.isNamespaceImport(node.importClause.namedBindings)
+    ) {
+      const ident = node.importClause.namedBindings.name.text;
+      const spec = (node.moduleSpecifier as ts.StringLiteral).text;
+      nameSpaceImports.set(ident, spec);
     }
-  );
-
-  if (!componentMapSourceFile) throw ReferenceError(`Failed to find file ${componentMapFullPath}`);
-  // this map matches all raw import strings (i.e. * as component) to import strings
-  const importStringsMap: Record<string, string> = {};
-  // this map will match component names only to full resolved source file paths
-  const componentImportsMap: Map<string, string> = new Map();
-
-  let mapExportName: string = '';
-  // all new xyz() statements in file
-  const newAssignments: ts.Node[] = [];
-  // all map.set() assignments in file
-  const mapAssignments: ts.CallExpression[] = [];
-
-  // this function will traverse the  map = new Map([/values/]) statement
-  // and get the component names registered in map from it
-  const traverseNewStatement = (node: ts.Node) => {
-    // going through map invocation, we're looking for outer array value
-    if (ts.isArrayLiteralExpression(node)) {
-      ts.forEachChild(node, (childNode) => {
-        // and then parse each individual array pair (i.e. ['MyComp', MyComp])
-        if (!ts.isArrayLiteralExpression(childNode)) {
-          return;
-        }
-        const componentKey = childNode.elements[1].getText();
-        const componentImport = Object.keys(importStringsMap).find((importStatement) => {
-          const matcher = new RegExp(`\\b(${componentKey})\\b`);
-          return importStatement.match(matcher) !== null;
-        });
-        if (componentImport) {
-          const componentValue = importStringsMap[componentImport];
-          componentImportsMap.set(componentKey, componentValue);
-        }
-      });
-    } else if (node.getChildCount() > 0) {
-      ts.forEachChild(node, (childNode) => {
-        traverseNewStatement(childNode);
+  });
+
+  // 2) Find the map literal: new Map([...]) and collect entries
```

Comment:
you also need to look for map.set() calls. Nothing prevents users from adding entries outside of the `new` call